src/openfermion/utils/_lcu_util.py

```diff
@@ -0,0 +1,181 @@
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+
+"""Utility methods for LCU circuits."""
+
+from __future__ import absolute_import
+from __future__ import division
+
+import math
+
+
+def _partial_sums(vals):
+    total = 0
+    for v in vals:
+        yield total
+        total += v
+    yield total
+
+
+def _differences(weights):
+    p = None
+    have_p = False
+    for w in weights:
+        if have_p:
+            yield w - p
+        p = w
+        have_p = True
+
+
+def _discretize_probability_distribution(unnormalized_probabilities, epsilon):
+    """Approximates probabilities with integers over a common denominator.
+
+    Args:
+        unnormalized_probabilities: A list of non-negative numbers proportional
+            to probabilities from a probability distribution. The numbers may
+            not be normalized (they do not have to add up to 1).
+        epsilon: The absolute error tolerance.
+
+    Returns:
+        A tuple containing:
```

Comment:
We can discuss here whether this is a poor practice, but in other parts of our code it has been common for us to talk about functions just returning two different objects. For instance, we would write under the returns line:

numerators_list(list): a list of of floats giving the numerators.
common_denominator(float): the common denominator.

unless you have a specific reason, I think this would be preferred just for consistency with other parts of the library. I'm not actually sure that sphinx will link the docs correctly if you do it your way.

Reply:
Done